internal/provider/resource_local_sensitive_file_test.go

```diff
@@ -0,0 +1,207 @@
+		resource "local_sensitive_file" "file" {
+		  source = "source_file"
+		  filename = "new_file"
+		}
+	`
+
+	r.UnitTest(t, r.TestCase{
+		Providers: testProviders,
+		Steps: []r.TestStep{
+			{
+				Config: config,
+				Check: func(s *terraform.State) error {
+					content, err := ioutil.ReadFile("new_file")
+					if err != nil {
+						return fmt.Errorf("config:\n%s\n,got: %s\n", config, err)
+					}
+					if string(content) != source_content {
+						return fmt.Errorf("config:\n%s\ngot:\n%s\nwant:\n%s\n", config, content, source_content)
+					}
+					return nil
+				},
+			},
+		},
+		CheckDestroy: func(*terraform.State) error {
+			if _, err := os.Stat("new_file"); os.IsNotExist(err) {
+				return nil
+			}
+			return errors.New("local_sensitive_file did not get destroyed")
+		},
+	})
+}
+
+func TestLocalSensitiveFile_Permissions(t *testing.T) {
+	td := createTestingTempDir(t)
+	defer os.RemoveAll(td)
+
+	destinationDirPath := td
+	destinationFilePath := filepath.Join(destinationDirPath, "local_sensitive_file")
+	destinationFilePath = strings.ReplaceAll(destinationFilePath, `\`, `\\`)
+	filePermission := os.FileMode(0600)
+	directoryPermission := os.FileMode(0700)
+	skipDirCheck := false
+	config := fmt.Sprintf(`
+		resource "local_sensitive_file" "file" {
+			content              = "This is some content"
+			filename             = "%s"
+			file_permission      = "0600"
+			directory_permission = "0700"
+		}`, destinationFilePath,
+	)
+
+	r.UnitTest(t, r.TestCase{
+		Providers: testProviders,
+		Steps: []r.TestStep{
+			{
+				Config: config,
+				PreConfig: func() {
+					// if directory already existed prior to check, skip check
+					if _, err := os.Stat(path.Dir(destinationFilePath)); !os.IsNotExist(err) {
+						skipDirCheck = true
+					}
+				},
+				Check: func(s *terraform.State) error {
+					if runtime.GOOS == "windows" {
+						// skip all checks if windows
+						return nil
+					}
+
+					fileInfo, err := os.Stat(destinationFilePath)
+					if err != nil {
+						return fmt.Errorf("config:\n%s\ngot:%s\n", config, err)
+					}
+
+					if fileInfo.Mode() != filePermission {
+						return fmt.Errorf(
+							"File permission.\nconfig:\n%s\nexpected:%s\ngot: %s\n",
+							config, filePermission, fileInfo.Mode())
+					}
+
+					if !skipDirCheck {
+						dirInfo, _ := os.Stat(path.Dir(destinationFilePath))
+						// we have to use FileMode.Perm() here, otherwise directory bit causes issues
+						if dirInfo.Mode().Perm() != directoryPermission.Perm() {
+							return fmt.Errorf(
+								"Directory permission.\nconfig:\n%s\nexpected:%s\ngot: %s\n",
+								config, directoryPermission, dirInfo.Mode().Perm())
+						}
+					}
+
+					return nil
+				},
+			},
+		},
+		CheckDestroy: func(*terraform.State) error {
+			if _, err := os.Stat(destinationFilePath); os.IsNotExist(err) {
+				return nil
+			}
+			return errors.New("local_sensitive_file did not get destroyed")
+		},
+	})
+
+	defer os.Remove(destinationDirPath)
+
```

Comment:
I believe this is extraneous since it should be covered by `defer os.RemoveAll(td)` at the beginning.

```suggestion
```